pkg/controller/node/node_controller.go

```diff
@@ -929,16 +1065,17 @@ func (ctrl *Controller) setDesiredMachineConfigAnnotation(nodeName, currentConfi
 			return err
 		}
 
-		newNode := oldNode.DeepCopy()
-		if newNode.Annotations == nil {
-			newNode.Annotations = map[string]string{}
-		}
-
-		if newNode.Annotations[daemonconsts.DesiredMachineConfigAnnotationKey] == currentConfig {
+		lns := ctrlcommon.NewLayeredNodeState(oldNode)
```

Comment:
Same question as above, we're denoting "layered node state" as the new consolidated way to mark node state?

Reply:
Yes. In the future, I'd like to port this to the MCD as well so that we have a singular place for these transitions and interrogations of the node object. That will also ensure that we can easily and rapidly test for edge-cases.